scripts/Muon/GUI/Common/fitting_widgets/tf_asymmetry_fitting/tf_asymmetry_fitting_model.py

```diff
@@ -0,0 +1,964 @@
+        for name in dataset_names:
+            fit_function = self.get_single_fit_function_for(name)
+            if fit_function is not None:
+                self._set_fit_function_parameter_values(fit_function, parameter_values)
+
+        if self.tf_asymmetry_mode:
+            self._update_tf_fit_function_parameters_for_single_fit(dataset_names, parameter_values)
+
+    def _update_tf_fit_function_parameters_for_single_fit(self, dataset_names: list, parameter_values: list):
+        """Updates the tf asymmetry function parameters for the given dataset names if in single fit mode."""
+        for name in dataset_names:
+            if name in self.dataset_names:
+                tf_single_function = self.get_tf_asymmetry_single_function(self.dataset_names.index(name))
+                self._set_fit_function_parameter_values(self._get_normal_fit_function_from(tf_single_function),
+                                                        parameter_values)
+
+    def _update_fit_function_parameters_for_simultaneous_fit(self, dataset_names: list, parameter_values: list) -> None:
+        """Updates the function parameters for the given dataset names if in simultaneous fit mode."""
+        self._set_fit_function_parameter_values(self.simultaneous_fit_function, parameter_values)
+
+        if self.tf_asymmetry_mode:
+            self._update_tf_fit_function_parameters_for_simultaneous_fit(dataset_names, parameter_values)
+
+    def _update_tf_fit_function_parameters_for_simultaneous_fit(self, dataset_names: list, parameter_values: list):
+        """Updates the tf asymmetry function parameters for the given dataset names if in simultaneous fit mode."""
+        number_parameters_per_domain = int(len(parameter_values) / len(dataset_names))
+        for name in dataset_names:
+            if name in self.dataset_names:
+                self._set_parameter_values_in_tf_asymmetry_simultaneous_function_domain(
+                    self.tf_asymmetry_simultaneous_function, self.dataset_names.index(name), parameter_values,
+                    number_parameters_per_domain)
+
+    def get_fit_workspace_names_from_groups_and_runs(self, runs: list, groups_and_pairs: list) -> list:
+        """Returns the workspace names to use for the given runs and groups/pairs."""
+        workspace_names = []
+        for run in runs:
+            for group_or_pair in groups_and_pairs:
+                workspace_names += self._get_workspace_name_from_run_and_group_or_pair(run, group_or_pair)
+        return workspace_names
+
+    def _get_workspace_name_from_run_and_group_or_pair(self, run: str, group_or_pair: str) -> list:
+        """Returns the workspace name to use for the given run and group/pair."""
+        if check_phasequad_name(group_or_pair) and group_or_pair in self.context.group_pair_context.selected_pairs:
+            return [get_pair_phasequad_name(self.context, group_or_pair, run, not self.fit_to_raw)]
```

Comment:
I agree, this method comes from the previous code that existed. However it uses the context held by the model and so I am not sure where this should go. It might be a good idea just to leave this for now and deal with it after the refactor